src/context.rs

```diff
@@ -12,15 +12,16 @@ use std::io::Error;
 use std::net::SocketAddr;
 use std::net::ToSocketAddrs;
 use std::sync::{ Arc, Mutex };
+use ws;
 
 // The `global` context available to all.
 pub struct Context {
     pub verbose: bool,
-
-    hostname: String,
-    http_port: u16,
-    ws_port: u16,
-    services: HashMap<String, Box<Service>>
+    pub hostname: String,
+    pub http_port: u16,
+    pub ws_port: u16,
+    pub services: HashMap<String, Box<Service>>,
+    pub sockets: HashMap<ws::util::Token, ws::Sender>,
```

Comment:
it still looks like `services` and `sockets` can be private. I'd rather keep that.